### DISCUSSION TITLE
Import of module linked against boost-python3 fails

### DISCUSSION THREAD

**Discussion by LutzGross** [Homebrew/discussions#5556] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: fa53e7b1e51a2deb7ec5a1e12452a1182dc342f7
Last commit: 2 days ago
Core tap JSON: 14 Aug 03:28 UTC
Core cask tap JSON: 14 Aug 01:38 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I am using boost-python3 to interface some C++ libraries - this worked with no problems. After the recent `brew upgrade` and rebuild of the backend libraries direct import of the interface library from python3 fails now:

Traceback (most recent call last):  File "<stdin>", line 1, in <module>
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.12/site-packages/numpy/core/_multiarray_umath.py", line 44, in __getattr__
    raise ImportError(msg)
ImportError: 
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.1 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

ImportError: numpy.core._multiarray_umath failed to import

**Comment by SMillerDev** [comment]
Do you have a way to reproduce this? The given error seems like it could come from anywhere.

**Reply by LutzGross** [reply]
Not so easy as it is a bigger code: https://github.com/LutzGross/esys-escript.github.io . Unfortunately its documentation is not in a good shape (happy to help if you wan to take install it).
  
The library that causes the problem is libescriptcpp.so which I am trying to import from python3.12

from esys.escriptcore.escriptcpp import *

(only statement). It is linked against boost-python3 and trilinos which I have rebuilt after `brew upgrade` (but it has no numpy or boost dependence)

I am running a very lean homebrew environment with

python@3.12: stable 3.12.5 
boost: stable 1.85.0 
boost-python3: stable 1.85.0 
numpy: stable 2.0.1
scipy: stable 1.14.0
python-matplotlib: stable 3.9.2 

Again this has worked before a recent upgrade (I guess when numpy moved to version2 ).

**Comment by cho-m** [comment, marked as answer]
If it is due to `boost-python3`, then may be fixed after https://github.com/Homebrew/homebrew-core/pull/181332

**Reply by LutzGross** [reply]
Upgrade to boost 1.86 solved the problem. Thanks.
